bdk_core/tests/test_sparse_chain.rs

```diff
@@ -149,7 +149,7 @@ fn checkpoint_limit_is_applied() {
     }
 
     assert_eq!(chain.iter_tx().count(), 10);
-    assert_eq!(chain.iter_checkpoints(..).count(), 5);
+    assert_eq!(chain.iter_checkpoints(..).count(), 10);
```

Comment:
I would actually try and preserve the checkpoint limit here unless it really gets in the way.